### PR TITLE
Upgrade to React 15

### PR DESCRIPTION
**Pull request by martijnrusschen**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,8 +33,8 @@
     "tether": "^1.1.1"
   },
   "peerDependencies": {
-    "react": "^0.14.0",
-    "react-dom": "^0.14.0"
+    "react": "^15.0.1",
```
Comment (rafeememon) [inline package.json]: Maybe this should be `^15.0.0` instead, I don't think this should have a preference for patch version

Also, should probably consider being able to support both 0.14 and 15.0, not sure if @souporserious has a preference?

**Comment by rafeememon** [comment]
@martijnrusschen, this library uses a lot of unstable React APIs.  Did you verify that all of them still work with v15.0?

**Comment by martijnrusschen** [comment]
Looks like there are two dependencies that need to be updated too:
- https://github.com/chenglou/react-motion/pull/320
- https://github.com/souporserious/react-motion-ui-pack

**Comment by gjuchault** [comment]
Hi guys, looking forward for this to be accepted.

I have updated to react v15 here is an error I have :

```
invariant.js:39 Uncaught Invariant Violation: traverseParentPath(...): Cannot traverse from and to the same ID, ``.invariant @ invariant.js:39traverseParentPath @ ReactInstanceHandles.js:157ReactInstanceHandles.traverseAncestors @ ReactInstanceHandles.js:285findDeepestCachedAncestor @ ReactMount.js:234ReactMount.findComponentRoot @ ReactMount.js:708ReactMount.findReactNodeByID @ ReactMount.js:679getNodeFromInstance @ ReactMount.js:180findDOMNode @ findDOMNode.js:43componentDidMount @ TetherComponent.js:62_assign.notifyAll @ CallbackQueue.js:66ON_DOM_READY_QUEUEING.close @ ReactReconcileTransaction.js:79Mixin.closeAll @ Transaction.js:202Mixin.perform @ Transaction.js:149batchedMountComponentIntoNode @ ReactMount.js:124Mixin.perform @ Transaction.js:136ReactDefaultBatchingStrategy.batchedUpdates @ ReactDefaultBatchingStrategy.js:63batchedUpdates @ ReactUpdates.js:97ReactMount._renderNewRootComponent @ ReactMount.js:277ReactMount__renderNewRootComponent @ ReactPerf.js:66ReactMount._renderSubtreeIntoContainer @ ReactMount.js:354ReactMount.render @ ReactMount.js:374React_render @ ReactPerf.js:66(anonymous function) @ client.js:22(anonymous function) @ main.bundle.js:342(anonymous function) @ main.bundle.js:343__webpack_require__ @ bootstrap 3570adf…:585fn @ bootstrap 3570adf…:109(anonymous function) @ main.bundle.js:9__webpack_require__ @ bootstrap 3570adf…:585webpackJsonpCallback @ bootstrap 3570adf…:21(anonymous function) @ main.bundle.js:1```

**Comment by souporserious** [comment]
Thanks for this. Can you change to `"react": "^0.14.0 || ^15.0.0"` please. Should be able to allow both right?

**Comment by gjuchault** [comment]
Could you please also publish the package on npm ? Thanks!

**Pull request merged**